app/src/main/java/com/example/moviewithfragments/api/MovieApi.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.moviewithfragments.api
+
+
+import com.example.moviewithfragments.BuildConfig
+import com.example.moviewithfragments.model.Result
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface MovieApi {
+    @GET("movie/popular")
+    suspend fun getMovies(): Response<Result>
+
+    @GET("discover/movie")
+    suspend fun searchMovieByYear(
+        @Query("primary_release_year") year: Int?
+    ): Response<Result>
+
+    @GET("search/movie")
+    suspend fun searchMovie(
+        @Query("query") query: String?
+    ): Response<Result>
+
+    companion object {
+
+        fun create(): MovieApi {
+            val client = OkHttpClient.Builder()
+                .addInterceptor { chain -> return@addInterceptor addApiKeyToRequests(chain) }
+                .build()
+
+            return Retrofit.Builder()
+                .baseUrl(MovieApiUtilities.BASE_URL)
+                .client(client)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+                .create(MovieApi::class.java)
```

Comment:
extract client and API into different functions and use single client only.
This will create new client every time we want to get MovieAPI